tests/network/netwrok_integration_test.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package network
+
+import (
+	cfg "github.com/ChainSafe/gossamer/config"
+	"github.com/ChainSafe/gossamer/dot/rpc/modules"
+	"github.com/ChainSafe/gossamer/lib/common"
+	libutils "github.com/ChainSafe/gossamer/lib/utils"
+	"github.com/ChainSafe/gossamer/tests/utils"
+	"github.com/ChainSafe/gossamer/tests/utils/config"
+	"github.com/ChainSafe/gossamer/tests/utils/node"
+	"github.com/ChainSafe/gossamer/tests/utils/retry"
+	"github.com/ChainSafe/gossamer/tests/utils/rpc"
+	"github.com/adrg/xdg"
+	"github.com/stretchr/testify/require"
+
+	"context"
+	"os"
+	"testing"
+	"time"
+)
+
+func TestKadDHTNetworkDiscovery(t *testing.T) {
+	if utils.MODE != "network" {
+		t.Skip("RPC tests are disabled, going to skip.")
+	}
+
+	genesisPath := libutils.GetWestendDevRawGenesisPath(t)
+	con := config.Default()
+	con.ChainSpec = genesisPath
+	con.Core.Role = common.FullNodeRole
+	con.RPC.Modules = []string{"system", "author", "chain"}
+	con.Network.MinPeers = 1
+	con.Network.MaxPeers = 20
+	con.Network.NoMDNS = true // Turning off mDNS, purpose of this test is purly use only kadDHT discovery
+	con.Core.BabeAuthority = true
+	con.Log.Sync = "trace"
+	con.Network.Port = 7001
+	// ID: 12D3KooWMHixgmjFYM4VyQNDTKMvN9BPw47Tyyb6LPZ43EavV68m
+	con.BasePath = xdg.DataHome + "/gossamer/westend-local/alice"
```

Comment:
I would suggest to use `t.TempDir()` to create an ephemeral dir where we can test